File: lib/cli/commands/hooks.mjs
import { parseArgs } from 'node:util';
import { EXIT, NemesisError } from '../../errors.mjs';

// PR05: surface only — hooks are implemented in PR38.
export function runHooks(argv, { stdout, stderr, cwd }) {
  const parsed = parseArgs({ args: argv, allowPositionals: true, strict: false });
  const [action, target] = parsed.positionals;
  if (!action || !['install', 'uninstall'].includes(action)) {
    throw new NemesisError('hooks requires install|uninstall [--pre-commit|--pre-push]', { code: 'USAGE', exitCode: EXIT.USAGE });
  }
  stdout.write(`${JSON.stringify({
    schemaVersion: 1, kind: 'nemesis-hooks', action, target: target ?? null,
    implemented: false, note: 'hook generation lands in PR38',
  }, null, 2)}\n`);
  return { exitCode: EXIT.PASS };
}

File: engine/bins/legion/src/commands/hooks.rs
use super::CommandResult;
use crate::cli::CommonArgs;
use serde_json::json;

pub fn run(args: CommonArgs) -> CommandResult {
    let argv = args
        .args
        .iter()
        .map(|value| value.to_string_lossy().into_owned())
        .collect::<Vec<_>>();
    let action = argv.first().map(String::as_str);
    if !matches!(action, Some("install") | Some("uninstall")) {
        return Err(super::CommandError::usage(
            "hooks requires install|uninstall [--pre-commit|--pre-push]",
        ));
    }
    let target = argv.get(1).cloned();
    Ok(json!({
        "schemaVersion": 1,
        "kind": "legion-hooks",
        "action": action.unwrap(),
        "target": target,
        "implemented": false,
        "note": "hook generation lands in PR38",
    }))
}
